infrastructure/hub-and-spoke-peerings/main.tf

```diff
@@ -0,0 +1,325 @@
+  network              = module.vpc-spoke-2.name
+  admin_ranges_enabled = true
+  admin_ranges         = local.all_subnet_cidrs
+}
+
+##############################################################
+#                           DNS Zones                        #
+##############################################################
+
+module "hub-private-zone" {
+  source = "../../modules/dns"
+
+  project_id      = var.hub_project_id
+  type            = "private"
+  name            = "${var.private_dns_zone_name}-hub-private"
+  domain          = var.private_dns_zone_domain
+  client_networks = [module.vpc-hub.self_link]
+  recordsets = [
+    { name = "hub-test-vm", type = "A", ttl = 300, records = module.vm-hub.internal_ips },
+    { name = "spoke-1-test-vm", type = "A", ttl = 300, records = module.vm-spoke-1.internal_ips },
+    { name = "spoke-2-test-vm", type = "A", ttl = 300, records = module.vm-spoke-2.internal_ips },
+  ]
+}
+
+module "hub-forwarding-zone" {
+  source = "../../modules/dns"
+
+  project_id = var.hub_project_id
+  type       = "forwarding"
+  name       = "${var.forwarding_dns_zone_name}-hub-forwarding"
+  domain     = var.forwarding_dns_zone_domain
+
+  client_networks = [module.vpc-hub.self_link]
+  forwarders      = [module.on-prem-in-a-box.dns_ip_address]
+}
+
+module "spoke-1-peering-zone-to-hub-private-zone" {
+  source = "../../modules/dns"
+
+  project_id = var.spoke_1_project_id
+  type       = "peering"
+  name       = "${var.private_dns_zone_name}-spoke-1-peering-to-hub-private"
+  domain     = var.private_dns_zone_domain
+
+  client_networks = [module.vpc-spoke-1.self_link]
+  peer_network    = module.vpc-hub.self_link
+}
+
+module "spoke-2-peering-zone-to-hub-private-zone" {
+  source = "../../modules/dns"
+
+  project_id = var.spoke_2_project_id
+  type       = "peering"
+  name       = "${var.private_dns_zone_name}-spoke-2-peering-to-hub-private"
+  domain     = var.private_dns_zone_domain
+
+  client_networks = [module.vpc-spoke-2.self_link]
+  peer_network    = module.vpc-hub.self_link
+}
+
+module "spoke-1-peering-zone-to-hub-forwarding-zone" {
+  source = "../../modules/dns"
+
+  project_id = var.spoke_1_project_id
+  type       = "peering"
+  name       = "${var.private_dns_zone_name}-spoke-1-peering-to-hub-forwarding"
+  domain     = var.forwarding_dns_zone_domain
+
+  client_networks = [module.vpc-spoke-1.self_link]
+  peer_network    = module.vpc-hub.self_link
+}
+
+module "spoke-2-peering-zone-to-hub-forwarding-zone" {
+  source = "../../modules/dns"
+
+  project_id = var.spoke_2_project_id
+  type       = "peering"
+  name       = "${var.private_dns_zone_name}-spoke-2-peering-to-hub-forwarding"
+  domain     = var.forwarding_dns_zone_domain
+
+  client_networks = [module.vpc-spoke-2.self_link]
+  peer_network    = module.vpc-hub.self_link
+}
+
+##############################################################
+#           TESTING: Inbount DNS Forwarding                  #
```

Comment:
typo ;)

Reply:
This is a bit tricky: people needing a simple example to copy and modify will have to remove a lot of stuff here. Add to that that the inbound policy needs manual configuration (the address problem), and I would much prefer removing it entirely. It's a very specialized use case, and my thought is that people who need it know how it works and can add it, for all the other people it just makes every more complex.